learn2learn/algorithms/lightning/lightning_protonet.py

```diff
@@ -139,3 +146,19 @@ def meta_learn(self, batch, batch_idx, ways, shots, queries):
         eval_loss = self.loss(logits, query_labels)
         eval_accuracy = accuracy(logits, query_labels)
         return eval_loss, eval_accuracy
+
+    def cache_on_validate_step(self, batch, batch_idx):
+        data, labels = batch
+        embeddings = self.features(data)
+        for e, l in zip(embeddings, labels):
+            self.support.append(e)
+            self.support_labels.append(l)
```

Comment:
This is potentially confusing: we're accumulating support samples across different tasks so that we can fit the classifier on increasingly more validation data. Fine, so long as each task's samples are labelled consistently with previous tasks' labels (eg, dogs are all labelled as class # 4). But that's not always the case in few-shot classification, where dogs could be class # 4 in the first task, but class # 1 in the second. Is there a way to handle both scenarios?

Reply:
To add to @seba-1511's comment, I think we do not need to cache data and it can potentially be incorrect from an implementation point of view. Though, I might be missing something

Reply:
Oh interesting, I didn't think about this @seba-1511. 
@pietrolesci We are caching the values, as when doing production inference, we would get some labelled data and want to use them all to compute the protypes.
I believe we should prevent the transforms to make random class assignments when doing predictions.

Reply:
Let's drop inference for now then, as it doesn't seem mature yet.